### PR TITLE
Cut down builder

### PR DESCRIPTION
**Pull request by madig**
Reduces OutlineBuilder to the minimum and moves the responsibilities of GlpyhBuilder back into the parser.

Closes https://github.com/linebender/norad/issues/95.

### CODE REVIEW

File: src/glyph/parse.rs [hunk 1]
```diff
@@ -46,43 +50,91 @@ impl<'names> GlifParser<'names> {
         raw_xml: &[u8],
         buf: &mut Vec<u8>,
     ) -> Result<Glyph, GlifLoadError> {
+        let mut seen_advance = false;
+        let mut seen_image = false;
+        let mut seen_lib = false;
+        let mut seen_note = false;
+        let mut seen_outline = false;
```
Comment (cmyr) [inline src/glyph/parse.rs:57]: can't we just check the various parts of the glyph object to see if they've been set?

Comment (madig) [reply src/glyph/parse.rs:57]: Well, outline, lib and advance.(height|width) aren't using Option, so we can't tell apart not-yet-encountered and default-value-of-zero-or-empty... but yes for image and note.

File: src/glyph/parse.rs [hunk 2]
```diff
@@ -46,43 +50,87 @@ impl<'names> GlifParser<'names> {
...
-                        "lib" => self.parse_lib(reader, raw_xml, buf)?, // do this at some point?
-                        "note" => self.parse_note(reader, buf)?,
-                        _other => return Err(ErrorKind::UnexpectedTag.into()),
+                Event::Start(start) => match start.name() {
+                    b"outline" => {
+                        if seen_outline {
+                            return Err(ErrorKind::DuplicateElement("outline").into());
+                        }
+                        seen_outline = true;
+                        self.parse_outline(reader, buf)?
```
Comment (cmyr) [inline src/glyph/parse.rs:66]: is this clearer as,

```suggestion
                    b"outline" if seen_outline => return Err(ErrorKind::DuplicateElement("outline").into()),
                    b"outline" => {
                        seen_outline = true;
                        self.parse_outline(reader, buf)?
                    }
                        
```

**Review by cmyr — APPROVED** [review]
okay one more little note that you can take or leave, it applies to all these match statements; I find that sometimes using two matches with the conditional inline is clearer than having the conditional inside the match block?

File: src/glyph/parse.rs (hunk 2 above)
Comment (madig) [reply src/glyph/parse.rs]: Let's try.

**Pull request merged**